analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+    scales = "free_y"
+  ) +
+  labs(y = "Average silhouette width")
+```
+
+```{r}
+# cluster purity for different params
+ggplot(summary_stats_df, aes(x = nn_param, y = mean_purity)) +
+  geom_point() +
+  facet_wrap(vars(library_id),
+    scales = "free_y"
+  ) +
+  labs(y = "Average cluster purity")
+```
+
+Here we see that both silhouette width and cluster purity peak around k=20 or k=25. 
+For `SCPCL000822`, increasing k after 25 does not change the width or purity, 
+For `SCPCL000824`, increasing k after 20-25 decreases the width and keeps purity around the same. 
+
+```{r, warning=FALSE, message=FALSE}
+# calculate cluster stability
+stability_df <- sce_list |>
+  purrr::imap(\(sce, id){
+    get_cluster_stability(sce, id, all_cluster_results)
+  }) |>
+  dplyr::bind_rows(.id = "library_id") |>
+  dplyr::mutate(
+    k_value = as.factor(k_value),
+    # make sure that k = 5 comes first
+    k_value = forcats::fct_relevel(k_value, "5", after = 0)
+  )
+
+# plot stability across all values of k
+ggplot(stability_df, aes(x = k_value, y = ari)) +
+  geom_point() +
+  facet_grid(rows = vars(library_id)) +
+  labs(title = "Cluster stability")
+```
+
+Here we see that cluster stability increases as k increases for both libraries. 
+We see a plateau start to appear around k = 20 and k = 25. 
+
+## Compare clusters to cell types from `SingleR`
+
+Now we will compare the clustering results to the cell type assignments obtained from `SingleR` in the `aucell-singler-annotation.sh` workflow. 
+To compare results we will calculate the Jaccard similarity index between clusters and cell types. 
+We expect that good clustering will line up with the cell type assignments so that there is ~ 1 cell type per cluster. 
+
+For the plots, we will only display the top 7 cell types and all other cells will be lumped together into `All remaining cell types`. 
+
+```{r}
+cluster_classification_df <- all_cluster_results |>
+  # add in classifications from singler
+  dplyr::left_join(classification_df, by = c("library_id", "barcodes")) |>
+  dplyr::mutate(
+    # first grab anything that is tumor and label it tumor
+    # NA should be unknown
+    singler_annotation = dplyr::case_when(
+      stringr::str_detect(singler_annotation, "tumor") ~ "tumor",
+      is.na(singler_annotation) ~ "unknown", # make sure to separate out unknown labels
+      .default = singler_annotation
+    ) |>
+      forcats::fct_relevel("tumor", after = 0),
+    # get the top cell types for plotting later
+    singler_lumped = singler_annotation |>
+      forcats::fct_lump_n(7, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf)
+  )
+```
+
+
+```{r, fig.height=10, fig.width=7}
+# get heatmap for each library showing cluster assignments vs. singler assignments
+names(sce_list) |>
+  purrr::map(\(id){
+    cluster_celltype_heatmap(
+      cluster_classification_df,
+      id
```

Comment:
need to set factors for this plot, 5 is last again.

Reply:
I wonder if we should turn off (at least y-axis) clustering for these heatmaps so that the cell labels are always in the same order. I think that will make it easier to interpret.